toolchain/check/context.h

```diff
@@ -265,8 +263,9 @@ class Context {
   // Prints a diagnostic for a duplicate name.
   auto DiagnoseDuplicateName(SemIRLoc dup_def, SemIRLoc prev_def) -> void;
 
-  // Prints a diagnostic for a poisoned name.
-  auto DiagnosePoisonedName(SemIRLoc loc) -> void;
+  // Prints a diagnostic for a poisoned name: Name used before declared.
```

Comment:
It feels a bit like you're just copying diagnostic format here, but I think it'd help to instead phrase in terms of how the function is intended to be used. How about:
```suggestion
  // Prints a diagnostic for a poisoned name when it's later declared.
```

Reply:
Done.